v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/templates/PubSubToElasticsearch.java

```diff
@@ -53,32 +54,70 @@
+            + "- <code>NAMESPACE</code> is the value of the <code>namespace</code> template parameter, or <code>default</code> if not specified."
+      },
+      optionsClass = PubSubToElasticsearchOptions.class,
+      skipOptions = {
+        "index",
+        "pythonExternalTextTransformGcsPath",
+        "pythonExternalTextTransformFunctionName",
+      }, // Template just ignores what is sent as "index"
+      flexContainerName = "pubsub-to-elasticsearch",
+      documentation =
+          "https://cloud.google.com/dataflow/docs/guides/templates/provided/pubsub-to-elasticsearch",
+      contactInformation = "https://cloud.google.com/support",
+      preview = true,
+      requirements = {
+        "The source Pub/Sub subscription must exist and the messages must be encoded in a valid JSON format.",
+        "A publicly reachable Elasticsearch host on a Google Cloud instance or on Elastic Cloud with Elasticsearch version 7.0 or above. See <a href=\"https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/googlecloud-to-elasticsearch/docs/PubSubToElasticsearch/README.md#google-cloud-integration-for-elastic\">Google Cloud Integration for Elastic</a> for more details.",
+        "A Pub/Sub topic for error output.",
+      },
+      streaming = true,
+      supportsAtLeastOnce = true),
+  @Template(
+      name = "PubSub_to_Elasticsearch_Xlang",
+      category = TemplateCategory.STREAMING,
+      displayName = "Pub/Sub to Elasticsearch With Python UDFs",
+      type = Template.TemplateType.XLANG,
+      description = {
+        "The Pub/Sub to Elasticsearch template is a streaming pipeline that reads messages from a Pub/Sub subscription, executes a Python user-defined function (UDF), and writes them to Elasticsearch as documents. "
+            + "The Dataflow template uses Elasticsearch's <a href=\"https://www.elastic.co/guide/en/elasticsearch/reference/master/data-streams.html\">data streams</a> feature to store time series data across multiple indices while giving you a single named resource for requests. "
+            + "Data streams are well-suited for logs, metrics, traces, and other continuously generated data stored in Pub/Sub.\n",
+        "The template creates a datastream named <code>logs-gcp.DATASET-NAMESPACE</code>, where:\n"
+            + "- <code>DATASET</code> is the value of the <code>dataset</code> template parameter, or <code>pubsub</code> if not specified.\n"
+            + "- <code>NAMESPACE</code> is the value of the <code>namespace</code> template parameter, or <code>default</code> if not specified."
+      },
+      optionsClass = PubSubToElasticsearchOptions.class,
+      skipOptions = {
+        "index",
+        "javascriptTextTransformGcsPath",
+        "javascriptTextTransformFunctionName"
+      }, // Template just ignores what is sent as "index" and javascript udf as this is for python
```

Comment:
This is a good idea to skip unused UDF options in each template "flavor". I think the other templates  should follow this pattern, namely GCS_to_Elasticsearch and PubSub_Proto_to_BigQuery which do not appear to follow this pattern.